### PR TITLE
[CDAP-16891] Upgrading pipeline drafts

### PR DESCRIPTION
**Pull request by itsanudeep**
Cherrypick of #12357  and #12385 for develop

Opening an incompatible pipeline that's saved as draft would take users through the import workflow to fix any issues they have with their pipeline.

JIRA: https://issues.cask.co/browse/CDAP-16891
JIRA: https://issues.cask.co/browse/CDAP-17005

### CODE REVIEW

File: cdap-ui/app/hydrator/controllers/create/partials/pipelineupgrade-modal-ctrl.js [hunk 1]
```diff
@@ -15,7 +15,7 @@
  */
 
 angular.module(PKG.name + '.feature.hydrator')
-  .controller('PipelineUpgradeModalController', function ($scope, rPipelineConfig, HydratorUpgradeService, $rootScope, HydratorPlusPlusConfigStore, $state, DAGPlusPlusFactory, GLOBALS, HydratorPlusPlusLeftPanelStore) {
+  .controller('PipelineUpgradeModalController', function ($scope, rPipelineConfig, HydratorUpgradeService, $rootScope, HydratorPlusPlusConfigStore, $state, DAGPlusPlusFactory, GLOBALS, HydratorPlusPlusLeftPanelStore,rIsImport) {
```
Comment (jennac3) [inline cdap-ui/app/hydrator/controllers/create/partials/pipelineupgrade-modal-ctrl.js]: nit: Could we have a blank line like this? `, rIsImport) {`

File: cdap-ui/app/hydrator/controllers/create/create-studio-ctrl.js [hunk 2]
```diff
@@ -77,6 +77,9 @@ class HydratorPlusPlusStudioCtrl {
       let config = {};
       config.artifact = artifact;
       HydratorPlusPlusConfigActions.initializeConfigStore(config);
+      if (rConfig.upgrade){
```
Comment (jennac3) [inline cdap-ui/app/hydrator/controllers/create/create-studio-ctrl.js]: nit: Could we have a blank line like this? `if (rConfig.upgrade) {`

Comment (itsanudeep) [reply cdap-ui/app/hydrator/controllers/create/create-studio-ctrl.js]: Fixed.

File: cdap-ui/app/hydrator/controllers/create/partials/pipelineupgrade-modal-ctrl.js (hunk 1 above)
Comment (itsanudeep) [reply cdap-ui/app/hydrator/controllers/create/partials/pipelineupgrade-modal-ctrl.js]: Fixed.

**Comment by yukiej** [comment]
build: https://builds.cask.co/browse/CDAP-UDUT815

**Comment by yukiej** [comment]
e2e tests: https://builds.cask.co/browse/IT-UPD2183

**Review by yukiej — APPROVED** [review]

**Pull request merged**